llvm-external-projects/iree-dialects/test/Dialect/iree_linalg_ext/roundtrip.mlir

```diff
@@ -643,3 +643,30 @@ func.func @topk_tensor_optional(%input_values: tensor<20x10x8x4xf32>) -> (tensor
 //  CHECK-SAME:      outs(%[[OUT_VALUES]], %[[OUT_INDICES]]
 //       CHECK:      iree_linalg_ext.yield
 //       CHECK:   return %[[RESULT]]#0, %[[RESULT]]#1
+
+// -----
+
+// CHECK: func.func @relayout(
+func.func @relayout(%arg0: tensor<3x3xf32>, %arg1: tensor<3x3xf32>) -> tensor<3x3xf32> {
+  // CHECK: iree_linalg_ext.pack
+  %1 = iree_linalg_ext.pack %arg0 inner_tiles [1, 1] into %arg1 : (tensor<3x3xf32> tensor<3x3xf32>) -> tensor<3x3xf32>
+  return %1 : tensor<3x3xf32>
+}
+
+// -----
+
+// CHECK: func.func @relayout(
+func.func @relayout(%arg0: memref<3x3xf32>, %arg1: memref<3x3xf32>) {
+  // CHECK: iree_linalg_ext.pack
+  iree_linalg_ext.pack %arg0 inner_tiles [1, 1] into %arg1 : (memref<3x3xf32> memref<3x3xf32>)
+  return 
+}
+
+// -----
+
+// CHECK: func.func @relayout(
+func.func @relayout(%arg0: memref<3x3xf32>, %arg1: memref<3x3xf32>) {
+  // CHECK: iree_linalg_ext.pack
+  iree_linalg_ext.pack %arg0 inner_tiles [1, 1] interchange = [0, 1, 2, 3] into %arg1 : (memref<3x3xf32> memref<3x3xf32>)
```

Comment:
I'd expect `interchange` to be size of `inner_tiles`. I dont see much value in having interchange for the outer dimensions.

Also given the `inner_tiles` is `[1, 1]` , i'd expect the output shape to be `memref<3x3x1x1xf32>` Its extra dimensions, but the op does not need to be too smart about it?